### ISSUE TITLE
Add `new/1` and `new/2` to Map and MapSet (mirroring the Keyword variant)

### ISSUE THREAD

**Issue by vadim-moz** [elixir-lang/elixir#3855]
There is no simple way to define `MapSet` literals. I would propose to append `.new` definition:
```elixir
def new
  %MapSet{}
end
def new(enum)
  Enum.into(enum, %MapSet{})
end
```

So a literal would look like:
```elixir
  MapSet.new([1, 2, 3])
```

I can prepare PR to implement it (I am also open to suggestion what other modules might benefit from similar API).

**Comment by josevalim** [comment]
Hello @vadim-moz! We do have `MapSet.new`. Then you can use `Enum.into/2` as you mentioned. :)

**Comment by vadim-moz** [comment]
Hello, @josevalim: I propose to _extend_ `MapSet.new` definiton. Otherwise, I am at loss how to compactly write my example. (I found creating literals to be repeating pattern, which is unpleasant. Creating extension module `MapSetX` and defining new with parameter there seems confusing as well.)

**Comment by josevalim** [comment]
We have typically not added the `.new/1` variants because you can just use `Enum.into/2`. Although a lot of folks have been bit by it. I guess it wouldn't hurt to finally add `new/1` and `new/2` to Map and MapSet.

**Comment by josevalim** [comment]
Btw, the implementation should just call `put` directly. There is no need to go through Enum.into. :)

**Comment by vadim-moz** [comment]
I will prepare PR. Thank you very much!

**Comment by vadim-moz** [comment]
Also, can you clarify your mentioning of the `put`, please? The `MapSet.put` puts single value, and I am trying to insert multiples, so I am looking for `Enum` as a parameter. My understanding is that `Keyword` is in a bit special position because of syntactic sugar (there implicit list literal created from multiple parameters). Am I missing something?

**Comment by josevalim** [comment]
Instead of into, you can use:

    def new(collection) do
      Enum.reduce collection, %MapSet{}, &put(&2, &1)
    end

So you don't have the overhead of going through into every time (and also avoid the protocol dispatch).

**Comment by vadim-moz** [comment]
oh -- good idea. Thank you.

**Comment by lexmag** [comment]
Closed in #3919.
